vshard/cfg.lua

```diff
@@ -257,17 +288,23 @@ local function cfg_check_weights(weights)
     end
 end
 
-local function check_sharding(sharding)
-    local uuids = {}
+local function check_sharding(sharding, ctx)
+    local ids = {}
     local uris = {}
     local names = {}
     local is_all_weights_zero = true
-    local rebalancer_uuid
-    for replicaset_uuid, replicaset in pairs(sharding) do
-        if uuids[replicaset_uuid] then
-            error(string.format('Duplicate uuid %s', replicaset_uuid))
+    local rebalancer_id
+    if ctx.is_named then
+        -- It's allowed to have the same names for replicaset and replica.
+        ids.replicasets = {}
+        ids.replicas = {}
+    end
+    for replicaset_id, replicaset in pairs(sharding) do
+        local rs_ids = ctx.is_named and ids.replicasets or ids
```

Comment:
This line doesn't need to be in the loop. Same with `replica_ids`.

Reply:
Indeed, this logic is located before the loop now.